### ISSUE TITLE
ISSUE #5 - Create support thread for user

### ISSUE THREAD

**Issue by Shredmetal** [Shredmetal/olympus_discord_bot_python#16]
Need a way to create support thread for user

**Comment by Shredmetal** [comment]
notes for shred

```
import discord
from discord import app_commands
from discord.ext import commands

# ... (other imports remain the same)

# Define the roles that can create support threads for others
SUPPORT_CREATION_ROLES = ["Developer", "Olympian", "Administrator"]

def has_support_creation_permission(member: discord.Member) -> bool:
    return any(role.name in SUPPORT_CREATION_ROLES for role in member.roles)

def register_commands(bot):
    @bot.tree.command(name="support", description="Initiate an Olympus support request")
    @app_commands.describe(user="The user to create a support thread for (optional)")
    @commands.cooldown(1, 60, commands.BucketType.user)
    async def support(interaction: discord.Interaction, user: discord.Member = None):
        # If a user is specified and it's not the command invoker
        if user and user != interaction.user:
            # Check if the command invoker has permission to create threads for others
            if not has_support_creation_permission(interaction.user):
                await interaction.response.send_message(
                    "You don't have permission to create support threads for other users.",
                    ephemeral=True
                )
                return
        else:
            # If no user is specified or the user is the command invoker, use the command invoker
            user = interaction.user

        if interaction.channel_id != COMMUNITY_SUPPORT_CHANNEL_ID:
            # ... (rest of the channel check remains the same)

        # ... (permission and channel type checks remain the same)

        thread_name = f"Support for {user.name}"

        try:
            thread = await interaction.channel.create_thread(
                name=thread_name,
                auto_archive_duration=1440,
                type=discord.ChannelType.public_thread
            )
        except discord.Forbidden as e:
            # ... (error handling remains the same)

        await interaction.response.send_message(
            content=f"Support thread created for {user.mention}: {thread.mention}\n"
                    f"Please upload your log files (Olympus_log.txt and dcs.log) in this thread.",
            ephemeral=True
        )

        await thread.send(
            f"{user.mention} Welcome to your support thread. Before proceeding, please ensure you have read"
            # ... (rest of the message remains the same)
        )

        set_thread_state(thread.id, ThreadState.AWAITING_LOGS)

    # ... (error handler remains the same)

```

**Comment by Shredmetal** [comment]
resolved in PR#19
